### DISCUSSION TITLE
Current screen resolution

### DISCUSSION THREAD

**Discussion by DenisTukhvatshin** [AvaloniaUI/Avalonia#8952] [Q&A] [answered]
How to find out the current screen resolution.
FontFamily="Roboto" FontSize="{Binding Bounds.Size, RelativeSource={RelativeSource AncestorType=Window}, Converter={StaticResource ScreenSizeConverter}}"
I tried this, but it outputs the width and height set in the application, not the current resolution.

**Comment by DenisTukhvatshin** [comment, marked as answer]
```
        if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
            var size = desktop.MainWindow.Screens.Primary.Bounds.Size;
        }
```
That's how it happened

**Comment by DenisTukhvatshin** [comment]
in some cases, after contacting the converter, he has it
```
        if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
        {
            var size = desktop.MainWindow?.Screens.Primary.Bounds.Size;
//MainWindow is null here sometimes
        }
```

**Reply by maxkatz6** [reply]
It can happen when converted was called during main window initialization. I.e. before it actually was shown.
